File: Encapsulation and Polymorphism/02. BankSystem/Models/Customer.cs
namespace BankSystem.Models
{
    using System;
    using BankSystem.Contracts;

    public abstract class Customer : ICustomer
    {
        private string name;

        public Customer(string name)
        {
            this.Name = name;
        }

        public string Name
        {
            get
            {
                return this.name;
            }

            set
            {
                if (string.IsNullOrEmpty(value))
                {
                    throw new ArgumentNullException("Customer name cannot be null!");
                }

                this.name = value;
            }
        }
    }
}

File: OOP in Java/OneLevShop/src/models/Customer.java
package models;

public class Customer {
	private String name;
	private int age;
	private double balance;
	
	public Customer(String name, int age, double balance) throws Exception {
		this.setName(name);
		this.setAge(age);
		this.setBalance(balance);
	}
	
	public String getName() {
		return name;
	}
	
	public void setName(String name) throws Exception {
		if(name == null && name.isEmpty()){
			throw new Exception("Name cannot be null or empty!");			
		}
			
		this.name = name;
	}
	
	public int getAge() {
		return age;
	}
	
	public void setAge(int age) throws Exception {
		if (age < 0) {
			throw new Exception("Age cannot be negative number!");
		}
		
		this.age = age;
	}
	
	public double getBalance() {
		return balance;
	}
	
	public void setBalance(double balance) throws Exception {
		if (balance < 0) {
			throw new Exception("Age cannot be negative number!");
		}
		
		this.balance = balance;
	}
}
